crates/primitives/src/transaction/mod.rs

```diff
@@ -715,6 +715,7 @@ impl reth_codecs::Compact for Transaction {
             }
             #[cfg(feature = "optimism")]
             Self::Deposit(tx) => {
+                let tx: reth_optimism_codecs::TxDepositEncode<'_> = tx.into();
```

Comment:
Hey @mattsse, should I close this PR then? Maybe I'm missing something but I'm not sure I can move the codecs tests without making the `TxDeposit` struct public outside of its crate

Reply:
yes exactly, then the optimism feature can't be removed if this type stays in the compact crate, which creates a bad patter for further rollups

Reply:
this optimism feature would be harmless because this is doesn't interfere with any other code.
and it's a lot cleaner than introducing an intermediary type for no reason